BigQuery: 'Client.update_dataset' systest error, 412 PreconditionFailed

First error in [this CI run](https://circleci.com/gh/GoogleCloudPlatform/google-cloud-python/7126):

```python
_______________________ TestBigQuery.test_update_dataset _______________________

self = <tests.system.TestBigQuery testMethod=test_update_dataset>

    def test_update_dataset(self):
        dataset = self.temp_dataset(_make_dataset_id('update_dataset'))
        self.assertTrue(_dataset_exists(dataset))
        self.assertIsNone(dataset.friendly_name)
        self.assertIsNone(dataset.description)
        self.assertEquals(dataset.labels, {})
    
        dataset.friendly_name = 'Friendly'
        dataset.description = 'Description'
        dataset.labels = {'priority': 'high', 'color': 'blue'}
        ds2 = Config.CLIENT.update_dataset(
            dataset,
>           ('friendly_name', 'description', 'labels'))

tests/system.py:189: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
google/cloud/bigquery/client.py:387: in update_dataset
    headers=headers)
...    
        if not 200 <= response.status_code < 300:
>           raise exceptions.from_http_response(response)
E           PreconditionFailed: 412 PATCH https://www.googleapis.com/bigquery/v2/projects/precise-truck-742/datasets/update_dataset_7126_1531263324: Precondition Failed

../.nox/sys-2-7/lib/python2.7/site-packages/google/cloud/_http.py:293: PreconditionFailed
```

--- a/bigquery/tests/system.py
+++ b/bigquery/tests/system.py
@@ -187,10 +187,6 @@ class TestBigQuery(unittest.TestCase):
         self.assertEqual(got.friendly_name, 'Friendly')
         self.assertEqual(got.description, 'Description')
 
-    @pytest.mark.skip(reason=(
-        'update_dataset() is flaky '
-        'https://github.com/GoogleCloudPlatform/google-cloud-python/issues/'
-        '5588'))
     def test_update_dataset(self):
         dataset = self.temp_dataset(_make_dataset_id('update_dataset'))
         self.assertTrue(_dataset_exists(dataset))
@@ -360,10 +356,6 @@ class TestBigQuery(unittest.TestCase):
                        table.dataset_id == DATASET_ID)]
         self.assertEqual(len(created), len(tables_to_create))
 
-    @pytest.mark.skip(reason=(
-        'update_table() is flaky '
-        'https://github.com/GoogleCloudPlatform/google-cloud-python/issues/'
-        '5589'))
     def test_update_table(self):
         dataset = self.temp_dataset(_make_dataset_id('update_table'))
 
@@ -403,10 +395,6 @@ class TestBigQuery(unittest.TestCase):
         with self.assertRaises(PreconditionFailed):
             Config.CLIENT.update_table(table2, ['description'])
 
-    @pytest.mark.skip(reason=(
-        'update_table() is flaky '
-        'https://github.com/GoogleCloudPlatform/google-cloud-python/issues/'
-        '5589'))
     def test_update_table_schema(self):
         dataset = self.temp_dataset(_make_dataset_id('update_table'))
 

